CRM/Core/BAO/Note.php

```diff
@@ -145,7 +145,7 @@ public static function add(&$params, $ids = array()) {
 
     if (!empty($params['entity_table']) && $params['entity_table'] == 'civicrm_contact' && !empty($params['check_permissions'])) {
       if (!CRM_Contact_BAO_Contact_Permission::allow($params['entity_id'], CRM_Core_Permission::EDIT)) {
-        throw new CRM_Exception('Permission denied to modify contact record');
+        throw new CRM_Core_Exception('Permission denied to modify contact record');
```

Comment:
Not sure I understand how this is related to a.net?

Reply:
Ah because when I added the exception catcher phpstorm wasn't happy because the exceptions didn't match up with those documented so I traced the functions throwing exceptions into it & stdised / documented them